tools/publisher/src/subcommand/generate_version_manifest.rs

```diff
@@ -203,7 +203,8 @@ fn hash_crate(path: &Path) -> Result<String> {
     Ok(stdout.trim().into())
 }
 
-fn hash_model(projection: &SmithyBuildProjection) -> Result<String> {
+fn hash_models(projection: &SmithyBuildProjection) -> Result<String> {
+    // Must match `hashModels` in `CrateVersioner.kt`
     let mut hashes = String::new();
     for import in &projection.imports {
         hashes.push_str(&sha256::digest_file(import).context("hash model")?);
```

Comment:
how does this get used from the Smithy side vs. Rust side?

Reply:
The publisher tool's `generate-version-manifest` subcommand creates the `versions.toml` which includes the model hashes, where this Rust implementation is used.

The Gradle build system then reads the prior release's `versions.toml` and determines if models were changed by doing its own model hashing.

An alternative we could consider is creating a tool that does model hashing and use it from both.